data/test/scenarios/events-test_filters.cfg

```diff
@@ -230,3 +230,119 @@
         {RETURN ({VARIABLE_CONDITIONAL triggers equals 4})}
     [/event]
 )}
+
+{GENERIC_UNIT_TEST event_test_filter_wfl (
+    [event]
+        name=start
+        {VARIABLE triggers 0}
+    [/event]
+    [event]
+        name=side turn
+        first_time_only=no
+        filter_formula="turn_number = side_number"
+        {VARIABLE_OP triggers add 1}
+        {ASSERT ({VARIABLE_CONDITIONAL side_number equals $turn_number})}
+    [/event]
+    [event]
+        name=turn 3
+        {RETURN ({VARIABLE_CONDITIONAL triggers equals 2})}
+    [/event]
+    [event]
+        name=side turn
+        first_time_only=no
+        [end_turn][/end_turn]
+    [/event]
+)}
+
+{GENERIC_UNIT_TEST event_test_filter_wfl2 (
+    [event]
+        name=start
+        {VARIABLE triggers 0}
+        [do_command]
+            [move]
+                x=7,7,6,5,4,3,3,3
+                y=3,4,4,5,4,4,3,2
+            [/move]
+        [/do_command]
+        {RETURN ({VARIABLE_CONDITIONAL triggers equals 3})}
+    [/event]
+    [event]
+        name=enter hex
+        first_time_only=no
+        filter_formula="loc.x = loc.y"
+        {VARIABLE_OP triggers add 1}
+    [/event]
+)}
+
+{GENERIC_UNIT_TEST event_test_filter_lua_serializable (
+    [event]
+        name=start
+        {VARIABLE triggers 0}
+    [/event]
+    [event]
+        name=side turn
+        first_time_only=no
+        [filter_condition]
+            [lua]
+                code=<<return wesnoth.current.turn == wesnoth.current.side>>
+            [/lua]
+        [/filter_condition]
+        {VARIABLE_OP triggers add 1}
+        {ASSERT ({VARIABLE_CONDITIONAL side_number equals $turn_number})}
+    [/event]
+    [event]
+        name=turn 3
+        {RETURN ({VARIABLE_CONDITIONAL triggers equals 2})}
+    [/event]
+    [event]
+        name=side turn
+        first_time_only=no
+        [end_turn][/end_turn]
+    [/event]
+)}
+
+{GENERIC_UNIT_TEST event_test_filter_lua_dynamic (
+    [event]
+        name=preload
+        [lua]
+            code=<<
+                wml.variables.triggers = 0
```

Comment:
Should also test that an event with `first_time_only=true` only triggers once. This correctly succeeds, so the code under test works:

```
diff --git a/data/test/scenarios/events-test_actions.cfg b/data/test/scenarios/events-test_actions.cfg
index 75448bad0f2..1071f2b4493 100644
--- a/data/test/scenarios/events-test_actions.cfg
+++ b/data/test/scenarios/events-test_actions.cfg
@@ -43,12 +43,20 @@
         name=preload
         [lua]
             code=<<
-                wml.variables.triggers = 0
+                wml.variables.only_once = 0
+                wml.variables.should_repeat = 0
+                wesnoth.game_events.add{
+                    name = 'new turn',
+                    first_time_only = true,
+                    action = function()
+                        wml.variables.only_once = wml.variables.only_once + 1
+                    end
+                }
                 wesnoth.game_events.add{
                     name = 'new turn',
                     first_time_only = false,
                     action = function()
-                        wml.variables.triggers = wml.variables.triggers + 1
+                        wml.variables.should_repeat = wml.variables.should_repeat + 1
                     end
                 }
             >>
@@ -68,7 +76,9 @@
     [/event]
     [event]
         name=side 2 turn 2
-        {RETURN ({VARIABLE_CONDITIONAL triggers equals 2})}
+        {ASSERT ({VARIABLE_CONDITIONAL only_once equals 1})}
+        {ASSERT ({VARIABLE_CONDITIONAL should_repeat equals 2})}
+        {SUCCEED}
     [/event]
 )}
```

Reply:
I'm a little bit confused. Is this a diff that I should just apply? Or is the idea that this diff produces an additional test, so we have the version without the diff applied _and_ the version with the diff applied?

Reply:
I meant it as something that could be just applied. Splitting them into separate tests would be fine too.